ape_foundry/provider.py

```diff
@@ -164,6 +164,9 @@ def project_folder(self) -> Path:
 
     @property
     def uri(self) -> str:
+        if self.config.host:
+            if "https" not in self.config.host or "http" not in self.config.host:
+                self.config.host = "http://" + self.config.host  # type: ignore
```

Comment:
Couple things:

1. Default to `http` for localhost and 127.0.0.1 but `https` for all other values
2. You can avoid the type ignore by using the walrus operator on line 167:

```python
if host := self.config.host:
   ...
```

3. Instead of changing the config value here, could we only change it on the `_host` value (with a little refactoring)?